CMakeLists.txt

```diff
@@ -343,7 +343,9 @@ if (INTERN_BUILD_MOBILE AND NOT BUILD_CAFFE2_MOBILE)
   set(FEATURE_TORCH_MOBILE ON)
   set(NO_API ON)
   set(USE_FBGEMM OFF)
-  set(USE_PYTORCH_QNNPACK ON)
+  if (NOT USE_PYTORCH_QNNPACK)
```

Comment:
why add this if?

Reply:
because it'll over-write this param in iOS build script. See the change in `build_ios.sh` down below.

Reply:
I'm not sure if there is a way to differentiate unset variable v.s. OFF variable in cmake. I think even if you explicitly set USE_PYTORCH_QNNPACK to OFF below, "NOT USE_PYTORCH_QNNPACK" will evaluate to TRUE and it will be set to ON again. I haven't tested, though.
I was thinking of not exposing this USE_PYTORCH_QNNPACK flag to external developer to reduce complexity, as it is always ON for mobile builds. But if there is legit reason to disable it for certain mobile devices, maybe we should just remove set(USE_PYTORCH_QNNPACK ON) here.

Reply:
@ljk53 should be `if(NOT DEFINED USE_PYTORCH_QNNPACK)`. My mistake.